indykite/resource_ingest_pipeline.go

```diff
@@ -0,0 +1,163 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indykite
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/indykite/indykite-sdk-go/config"
+	configpb "github.com/indykite/indykite-sdk-go/gen/indykite/config/v1beta1"
+)
+
+//nolint:gosec // there are no secrets
+const (
+	ingestPipelineSourcesTypeKey       = "sources"
+	ingestPipelineOperationsTypeKey    = "operations"
+	ingestPipelineAppAgentTokenTypeKey = "app_agent_token"
+)
+
+var (
+	ingestPipelineAppAgentTokenRegex = regexp.MustCompile(`^[A-Za-z0-9-_]+?\.[A-Za-z0-9-_]+?\.[A-Za-z0-9-_]+?$`)
+	ingestPipelineAllowedOperations  = []string{
+		"OPERATION_UPSERT_NODE",
+		"OPERATION_UPSERT_RELATIONSHIP",
+		"OPERATION_DELETE_NODE",
+		"OPERATION_DELETE_RELATIONSHIP",
+		"OPERATION_DELETE_NODE_PROPERTY",
+		"OPERATION_DELETE_RELATIONSHIP_PROPERTY",
+	}
+)
+
+func resourceIngestPipeline() *schema.Resource {
+	readContext := configReadContextFunc(resourceIngestPipelineFlatten)
+
+	return &schema.Resource{
+		Description: `Ingest pipeline configuration adds support for 3rd party data sources, which can be used to ingest data to the IndyKite system. The configuration also allows you to define the allowed operations in the pipeline.`,
+
+		CreateContext: configCreateContextFunc(resourceIngestPipelineBuild, readContext),
+		ReadContext:   readContext,
+		UpdateContext: configUpdateContextFunc(resourceIngestPipelineBuild, readContext),
+		DeleteContext: configDeleteContextFunc(),
+		Importer: &schema.ResourceImporter{
+			StateContext: basicStateImporter,
+		},
+
+		Timeouts: defaultTimeouts(),
+		Schema: map[string]*schema.Schema{
+			locationKey:   locationSchema(),
+			customerIDKey: setComputed(customerIDSchema()),
+			appSpaceIDKey: setComputed(appSpaceIDSchema()),
+
+			nameKey:        nameSchema(),
+			displayNameKey: displayNameSchema(),
+			descriptionKey: descriptionSchema(),
+			createTimeKey:  createTimeSchema(),
+			updateTimeKey:  updateTimeSchema(),
+
+			ingestPipelineSourcesTypeKey: {
+				Type:        schema.TypeList,
+				Description: "List of sources to be used in the ingest pipeline.",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				MinItems: 1,
+				MaxItems: 10,
+			},
+			ingestPipelineOperationsTypeKey: {
+				Type:        schema.TypeList,
+				Description: "List of operations which will be allowed to be used in the ingest pipeline.",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.All(
+						validation.StringInSlice(ingestPipelineAllowedOperations, false),
+					),
```

Comment:
I think you don't need `validation.All()` when you have single validation there

Reply:
Fixed